test/jdk/jdk/internal/vm/Continuation/BasicExp.java

```diff
@@ -0,0 +1,974 @@
+            assertEquals(frames, cont.isDone() ? List.of()
+                         : Arrays.asList("yield", "ord104_testMethod_dontinline", "ord103_testMethod_dontinline",
+                                         "ord102_testMethod_dontinline", "ord101_testMethod_dontinline", "run", "enter0", "enter"));
+        }
+    }
+
+    /**
+     * Deep recursion to exercise fast freezing into non-empty chunk
+     */
+    public static class ContinuationDeepRecursion extends TestCaseBase {
+        public int limit;
+        public int yield1_depth;
+        public int yield2_depth;
+
+        @Override
+        public void setup_dontjit(boolean warmup) {
+            if (warmup) {
+                limit = 10;
+                yield1_depth = 7;
+                yield2_depth = 3;
+            } else {
+                limit = 100;
+                yield1_depth = 70;
+                yield2_depth = 60;
+            }
+        }
+
+        @Override
+        public void run() {
+            log_dontjit("Continuation running on thread " + Thread.currentThread());
+            ord101_recurse_dontinline(0);
+        }
+
+        public void ord101_recurse_dontinline(int depth) {
+            if (depth >= limit) {
+                log_dontjit("yield at depth " + depth);
+                ord102_yield_dontinline(0);
+                log_dontjit("After yield at depth " + depth);
+                return;
+            }
+            ord101_recurse_dontinline(depth + 1);
+            if (depth == yield1_depth || depth == yield2_depth) {
+                log_dontjit("yield at depth " + depth);
+                ord102_yield_dontinline(0);
+                log_dontjit("After yield at depth " + depth);
+            }
+        }
+
+        // Add a few frames before yield
+        public void ord102_yield_dontinline(int depth) {
+            if (depth >= 2) {
+                Continuation.yield(THE_SCOPE);
+                return;
+            }
+            ord102_yield_dontinline(depth + 1);
+        }
+    }
+
+    /**
+     * Deep recursion to exercise fast freezing into non-empty chunk.
+     * nmethods have stack arguments.
+     */
+    public static class ContinuationDeepRecursionStackargs extends TestCaseBase {
+        public int limit;
+        public int yield1_depth;
+        public int yield2_depth;
+
+        @Override
+        public void setup_dontjit(boolean warmup) {
+            if (warmup) {
+                limit = 10;
+                yield1_depth = 7;
+                yield2_depth = 3;
+            } else {
+                limit = 100;
+                yield1_depth = 70;
+                yield2_depth = 60;
+            }
+        }
+
+        @Override
+        public void run() {
+            log_dontjit("Continuation running on thread " + Thread.currentThread());
+            long res = ord101_recurse_dontinline(0, i1, i2, i3, i4, i5, i6, i7, i8, i9, i10, i11);
+            if (res != i1+i2+i3+i4+i5+i6+i7+i8+i9+i10+i11) {
```

Comment:
Please fix the indentation. There are  several places were 'i1+i2' should be fixed to 'i1 + i2 '.

Reply:
Done